polus-file-renaming-plugin/src/main.py

```diff
@@ -0,0 +1,297 @@
+        prog="main",
+        description="""
+            Renames files using patterns, such as dd or c+"""
+        )
+    parser.add_argument(
+        "--inpDir",
+        dest = "inpDir", type=str, 
+        help="Input image collection to be processed by this plugin",
+        required=True
+        )
+    parser.add_argument(
+        "--outDir", dest="outDir", type=str, 
+        help="Output image collection of renamed files",
+        required=True
+        )
+    parser.add_argument(
+        "--filePattern", dest="filePattern", type=str, 
+        help="Filename pattern used to separate data",
+        required=True
+        )
+    parser.add_argument(
+        "--outFilePattern", dest="outFilePattern", type=str, 
+        help="Desired filename pattern used to rename and separate data",
+        required=True
+        )
+    #: Parse the arguments
+    args = parser.parse_args()
+    inpDir = args.inpDir
+    #: outDir is the new collection
+    outDir = args.outDir
+    #: Input pattern is the regex pattern input by user
+    filePattern = args.filePattern
+    #: Output pattern is regex pattern expected by user
+    outFilePattern = args.outFilePattern
+    
+    """Inputs"""
+    inp_files = list(Path(inpDir).iterdir())
+    inp_pattern = filePattern
+    out_pattern = outFilePattern
+    
+    """Process Inputs"""
+    #: Generate dict of named regular expression groups
+    regex_patterns = pattern_to_regex(inp_pattern)
+    #: Update file pattern to be regex-compatible, using dict of regex_patterns
+    inp = pattern_to_raw_f_string(inp_pattern, regex_patterns)
+    #: Convert out pattern to fstring to be used for str formatting
+    out_pattern_fstring = pattern_to_fstring(out_pattern)
+    #: List category names where input, output patterns are c, d)
+    category_list = replace_cat_label(inp_pattern, out_pattern)
+    #: List dict for each filename, capture groups, and corresponding matches
+    all_matches = gen_all_matches(inp, inp_files)
+    #: Iterate through matches
+    for i in range(0, len(all_matches)):
+        tmp_match = all_matches[i]
+        #: Convert numeric str dictionary values to int, if applicable
+        all_matches[i] = numstrvalue_to_int(tmp_match)
+    char_to_num = dict()
+    for c_to_d_category in category_list:
+        char_to_num[c_to_d_category] = non_numstr_value_to_int(
+            c_to_d_category, all_matches)
+    #: Convert tmp_match to integers, if applicable
+    for c_to_d_category in category_list:
+        for i in range(0, len(all_matches)):
+            if all_matches[i].get(c_to_d_category):
+                all_matches[i][c_to_d_category] = char_to_num[c_to_d_category][
+                    all_matches[i][c_to_d_category]]
+    for match in all_matches:
+        out_name = Path(outDir).resolve() / out_pattern_fstring.format(**match)
+        logger.info(f"Old file name {match['fname']} and new file name {out_name}")
+        #: Copy renamed file to output directory
```

Comment:
This looks good, but please move this to a `main` function with typed inputs. We are trying to only have input parsing done under the `if __name__=='__main__` block and then pass those parsed inputs to a `main` function. The reason is so that we can call this code from another Python function if we want.

Reply:
Thank you, that makes sense. I will move the needed parts to main()